common/plugins/com.liferay.ide.core/src/com/liferay/ide/core/AbstractLiferayNature.java

```diff
@@ -178,4 +171,8 @@ public void setProject( IProject project )
         this.currentProject = project;
     }
 
+    protected abstract String getNatureId();
+
+    protected abstract String[] getNatureIds();
```

Comment:
We dont' need both of these methods getNatureId() and getNatureIds().  I think that one Nature class would only have one nature, so lets remove the "getNatureIds()" method.